app/modules/services/FileSuggestionService/Sources/DefaultFileSuggestionService.swift

```diff
@@ -165,9 +165,16 @@ final class DefaultFileSuggestionService: FileSuggestionService {
     var files = [URL]()
     let addFileRef: (PBXFileReference) -> Void = { fileRef in
       if
-        let path = fileRef.path.map({ $0.resolvePath(from: projectDir) }),
+        var path = URL(string: fileRef.path ?? ""),
         !["app", "appex", "framework"].contains(path.pathExtension)
       {
+        var parent = fileRef.parent
+        while let parentPath = URL(string: parent?.path ?? "") {
+          path = path.resolve(from: parentPath)
+          parent = parent?.parent
+        }
+        path = path.resolve(from: projectDir)
+
```

Comment:
qq: if `fileRef.path` was to be `nil`, this would not work, correct? If so we should guard against that.

Reply:
It should still work because attempting to construct an URL with an empty String would result in `nil` (per the initializer's doc), and thus fail the `if` check.

Regardless, let me update to make it more explicit.